CONTRIBUTING.md

```diff
@@ -1,43 +1,122 @@
+If you are not satisfied with only reporting issues and want to get your hands dirty, then rest assured: Any help is appreciated! We are always thrilled to receive pull requests, and do our best to process them as fast as possible. Not sure if that typo is worth a pull request?
+**Do it!**
+**We will appreciate it.**
 
-Run jekyll in the root directory to build:
+The site uses [GitHub Pages](https://help.github.com/articles/what-are-github-pages/) with the site content in the `gh-pages` branch. To build the site locally, you will need to install Ruby, Bundler and Jekyll. If you've never worked with GitHub Pages before, just keep reading.
 
-	$ bundle exec jekyll build
+Or you already are a pro at working with GitHub Pages? You have set up the [git triangular workflow](https://www.sociomantic.com/blog/2014/05/git-triangular-workflow/) for this repository and have Ruby, Bundler and Jekyll running? Then what are you waiting for? Read the [contributing guidelines](#contributing-guidelines) section and start writing!
 
-You can also run a local server to test the site:
+Github pages can be created either with usual HTML or you can use [Jekyll](https://jekyllrb.com/), a static web site generator, which takes [Markdown](https://jekyllrb.com/docs/posts/) files and transforms them into a static webpage. Usually you want to write in markdown, as this is way more convenient than writing pure HTML.
 
-	$ bundle exec jekyll serve
+The following instructions are a shortened form of the official [GitHub article](https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/) on setting up your GitHub Pages site locally with Jekyll.
 
-Then go to [http://localhost:4000](http://localhost:4000). The pages are automatically updated when you edit a markdown file, though you do have to refresh the browser to see these changes. Eclipse later revisions have a decent markdown editor build in.
+### Install Git, Ruby and Bundler
 
-We use [Travis CI](https://travis-ci.org/osgi/osgi.enroute.site) to test build any pull requests.
+**Update/Install Git**
+* Check if you've already installed Git: type `git --version` in a console
+* If a version is printed out it's fine (at the time of writing:  2.14.1)
+* If not:
+ 	* **Windows:** go to [Git](https://git-scm.com/) and download/install the current version
+	* **Linux:** type `apt-get install git-all` in a console
 
-## Workflow
+**Update/Install Ruby**
+* Check if you've arleady installed Ruby: type `ruby --version` in a console
+* If a version is printed out it's fine (at the time of writing: 2.4.1)
+* If not:
+	* **Windows:** download [Ruby](https://rubyinstaller.org/) and install it
+	* **Linux:** type `apt-get install ruby` in a console
+> *There seems to be a problem with json 1.8.3 and Ruby 2.4.x. So you should either install 2.3.3 for now or follow these [instructions](https://github.com/github/pages-gem/issues/376)*
```

Comment:
Also macOS. XCode includes ruby 2.0 and I don't think we require a newer version than that. You can install ruby via brew also.